aviatrix/resource_aviatrix_remote_syslog.go

```diff
@@ -0,0 +1,179 @@
+			},
+			"excluded_gateways": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "List of excluded gateways.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"notls": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if not protected by TLS.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Enabled or not.",
+			},
+		},
+	}
+}
+
+func resourceAviatrixRemoteSyslogCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	remoteSyslog := &goaviatrix.RemoteSyslog{
+		Server:            d.Get("server").(string),
+		Port:              d.Get("port").(int),
+		Protocol:          d.Get("protocol").(string),
+		Index:             d.Get("index").(int),
+		Template:          d.Get("template").(string),
+		CaCertificate:     d.Get("ca_certificate_file_path").(string),
+		PublicCertificate: d.Get("public_certificate_file_path").(string),
+		PrivateKey:        d.Get("private_key_file_path").(string),
+	}
+
+	if !((remoteSyslog.CaCertificate != "" && remoteSyslog.PublicCertificate != "" && remoteSyslog.PrivateKey != "") ||
+		(remoteSyslog.CaCertificate == "" && remoteSyslog.PublicCertificate == "" && remoteSyslog.PrivateKey == "")) {
+		return fmt.Errorf("one or more certificates missing")
+	}
+
+	var excludeGateways []string
+	for _, v := range d.Get("excluded_gateways").(*schema.Set).List() {
+		excludeGateways = append(excludeGateways, v.(string))
+	}
+	if len(excludeGateways) != 0 {
+		remoteSyslog.ExcludeGatewayInput = strings.Join(excludeGateways, ",")
+	}
+
+	if err := client.EnableRemoteSyslog(remoteSyslog); err != nil {
+		return fmt.Errorf("could not enable remote syslog: %v", err)
+	}
+
+	d.SetId("remote_syslog_" + strconv.Itoa(remoteSyslog.Index))
+	return nil
+}
+
+func resourceAviatrixRemoteSyslogRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	server := d.Get("server").(string)
+
+	if server == "" {
+		id := d.Id()
+		log.Printf("[DEBUG] Looks like an import. Import Id is %s", id)
+		index, _ := strconv.Atoi(id[len(id)-1:])
+		d.Set("index", index)
+		d.SetId(id)
+	}
+
+	remoteSyslogStatus, err := client.GetRemoteSyslogStatus(d.Get("index").(int))
+	if err != nil {
+		return fmt.Errorf("could not get remote syslog status: %v", err)
+	}
+
+	idx, _ := strconv.Atoi(remoteSyslogStatus.Index)
+	d.Set("index", idx)
+	d.Set("server", remoteSyslogStatus.Server)
+	port, _ := strconv.Atoi(remoteSyslogStatus.Port)
+	d.Set("port", port)
+	d.Set("protocol", remoteSyslogStatus.Protocol)
+	d.Set("template", remoteSyslogStatus.Template)
+	d.Set("notls", remoteSyslogStatus.Notls)
+	d.Set("status", remoteSyslogStatus.Status)
+
+	var excludedGateways []interface{}
+	for _, v := range remoteSyslogStatus.ExcludedGateways {
+		excludedGateways = append(excludedGateways, v)
+	}
```

Comment:
I don't think it is necessary to transform the type to `[]interface{}`, you should be able to `d.Set` a slice of string no problem.